Content.Shared/GameObjects/Components/SharedStackComponent.cs

```diff
@@ -27,16 +21,25 @@ public abstract class SharedStackComponent : Component, ISerializationHooks
         private int _maxCount = 30;
 
         [ViewVariables(VVAccess.ReadWrite)]
-        public virtual int Count
+        [DataField("stackType", required:true, customTypeSerializer:typeof(PrototypeIdSerializer<StackPrototype>))]
+        public string StackTypeId { get; private set; } = string.Empty;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        public int Count
         {
             get => _count;
             set
             {
+                if (_count == value)
```

Comment:
Next step is to replace this with a ChangeStackEvent right?

Reply:
basically, yeah

Reply:
done now!